### PR TITLE
Delete work branches when done

### LINKED ISSUE #4523: [LakeFSOutputCommitter] Delete work branches when done
This should be configurable for job and for task branches separately.

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #4523.

### CODE REVIEW

**Comment by arielshaqed** [comment]
Review & merge this **after #4513**!

**Comment by arielshaqed** [comment]
Ready for review.

Manually tested.

File: clients/hadoopfs/src/main/java/io/lakefs/Constants.java
Comment (johnnyaug) [inline clients/hadoopfs/src/main/java/io/lakefs/Constants.java:12]: These configurations can both pass as a branch name configuration.
Would go with something as `committer.cleanup.delete_job_branch` - to clarify it's a boolean.

**Review by johnnyaug — APPROVED** [review]
Looks great! Small comment.

File: clients/hadoopfs/src/main/java/io/lakefs/Constants.java
Comment (arielshaqed) [reply clients/hadoopfs/src/main/java/io/lakefs/Constants.java]: We ended up with these names:

* `spark.hadoop.fs.lakefs.committer.delete_job_branch`
* `spark.hadoop.fs.lakefs.committer.delete_task_branch`

**Review by lynnro314 — APPROVED** [review]
Great, thanks!

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**
